processing/src/main/java/org/apache/druid/frame/key/FrameComparisonWidgetImpl.java

```diff
@@ -138,21 +139,29 @@ public RowKey readKey(int row)
   }
 
   @Override
-  public boolean isCompletelyNonNullKey(int row)
+  public boolean hasNonNullKeyParts(int row, int[] keyParts)
```

Comment:
Sidenote, we should mention somewhere in Javadoc that `compare` implementation assumes `null == null`, therefore this method can be called to filter out those rows.

Reply:
Added.